src/olympia/blocklist/mlbf.py

```diff
@@ -207,31 +208,44 @@ def hash_filter_inputs(cls, input_list):
             for (guid, version) in input_list
         ]
 
-    @property
-    def filter_path(self):
-        return self.storage.path('filter')
+    def filter_path(self, block_type: BlockType):
+        return self.storage.path(f'filter-{BaseMLBFLoader.data_type_key(block_type)}')
 
     @property
     def stash_path(self):
         return self.storage.path('stash.json')
 
-    def generate_and_write_filter(self):
+    def delete(self):
+        if self.storage.exists(self.storage.base_location):
+            self.storage.rm_stored_dir(self.storage.base_location)
+            log.info(f'Deleted {self.storage.base_location}')
+
+    def generate_and_write_filter(self, block_type: BlockType):
         stats = {}
 
+        blocked = self.data[block_type]
+
+        not_blocked = [
```

Comment:
Another thing I've noticed is that this generates a list of lists. I am not sure how that is encoded in the filter but I can see that the size of the filter is very small compared to what it used to be without this patch applied and, more serious, all add-ons seem to be blocked by Firefox with filters generated by this patch.

Let's flatten all that so that both `blocked` and `not_blocked` are plain lists of strings.

It isn't clear to me what this `verify()` method on the filter does, though, because we seem to have coverage for this `generate_and_write_filter` method.

Reply:
Try this maybe? That should address both comments above.

```diff
diff --git a/src/olympia/blocklist/mlbf.py b/src/olympia/blocklist/mlbf.py
index 26bd2264b6..14d51edf98 100644
--- a/src/olympia/blocklist/mlbf.py
+++ b/src/olympia/blocklist/mlbf.py
@@ -224,12 +224,16 @@ class MLBF:
         stats = {}
 
         blocked = self.data[block_type]
-
-        not_blocked = [
-            self.data[_block_type]
-            for _block_type in BlockType
-            if _block_type != block_type
-        ]
+        not_blocked = (
+            self.data.not_blocked_items
+            + self.data[
+                (
+                    BlockType.BLOCKED
+                    if block_type == BlockType.SOFT_BLOCKED
+                    else BlockType.SOFT_BLOCKED
+                )
+            ]
+        )
 
         bloomfilter = generate_mlbf(
             stats=stats,
```

Reply:
I think that might not be quite enough. But I've expanded to a loop to be a bit more clear. Also there was a bug there as we need to include not just the BlockTypes but also the not blocked items in general.